test/tests/functional/pbs_cpuset.py

```diff
@@ -140,17 +244,202 @@ def test_reliable_job_startup_on_cpuset(self):
+        # at which time the sister node is automatically released
+        # due to release_nodes_stageout=true set
+        self.server.delete(jid)
+
+        # Verify remaining job resources.
+        self.server.expect(JOB, {'job_state': 'E',
+                                 'Resource_List.mem': '2gb',
+                                 'Resource_List.ncpus': 1,
+                                 'Resource_List.select': self.job1_newsel,
+                                 'Resource_List.place': self.job1_place,
+                                 'Resource_List.nodect': 1,
+                                 'schedselect': self.job1_newsel,
+                                 'exec_host': self.job1_new_exec_host,
+                                 'exec_vnode': self.job1_new_exec_vnode},
+                           id=jid)
+        # Check various vnode status
+        attr0 = {'state': 'free', 'jobs': jid + '/0',
+                 'resources_assigned.ncpus': 1,
+                 'resources_assigned.mem': '2097152kb'}
+        self.server.expect(VNODE, attr0, id=self.n0)
+        attr1 = {'state': 'free', 'resources_assigned.ncpus': '0kb',
+                 'resources_assigned.mem': '0kb'}
+        for vn in [self.n1, self.n2, self.n3]:
+            self.server.expect(VNODE, attr1, id=vn)
+        # job's PBS_NODEFILE contents should match exec_host
+        ret = self.server.du.run_cmd(self.n0, cmd, sudo=False)
+        self.assertTrue(hostA in ret['out'])
+        self.assertFalse(hostB in ret['out'])
+        # Verify mom_logs
+        self.moms.values()[0].log_match(
+            "Job;%s;%s.+cput=.+ mem=.+" % (jid, self.n1), n=10, regexp=True)
+        self.moms.values()[1].log_match(
+            "Job;%s;DELETE_JOB2 received" % (jid,), n=20)
+        # Check account update ('u') record
+        msg0 = ".*%s;%s.*exec_host=%s" % ('u', jid, self.job1_exec_host_esc)
+        msg1 = ".*exec_vnode=%s" % self.job1_exec_vnode_esc
+        msg2 = ".*Resource_List\.mem=%s" % '6gb'
+        msg3 = ".*Resource_List\.ncpus=%d" % 9
+        msg4 = ".*Resource_List\.place=%s" % self.job1_place
+        msg5 = ".*Resource_List\.select=%s.*" % self.job1_sel_esc
+        msg = msg0 + msg1 + msg2 + msg3 + msg4 + msg5
+        self.server.accounting_match(msg=msg, regexp=True, n="ALL",
+                                     starttime=stime)
+        # Check to make sure 'c' (next) record got generated
+        msg0 = ".*%s;%s.*exec_host=%s" % ('c', jid, self.job1_new_exec_host)
+        msg1 = ".*exec_vnode=%s" % self.job1_new_exec_vnode_esc
+        msg2 = ".*Resource_List\.mem=%s" % '2097152kb'
+        msg3 = ".*Resource_List\.ncpus=%d" % 1
+        msg4 = ".*Resource_List\.place=%s" % self.job1_place
+        msg5 = ".*Resource_List\.select=%s.*" % self.job1_newsel
+        msg = msg0 + msg1 + msg2 + msg3 + msg4 + msg5
+        self.server.accounting_match(msg=msg, regexp=True, n="ALL",
+                                     starttime=stime)
```

Comment:
I thought you were also going to show that a job can use the released vnodes of the cpuset system?  Even though the job is still being staged out?

Reply:
The test will check after job is in stageout if self.n3 has status = free, resources_assigned.ncpus and resources_assigned.mem are 0 and that should be sufficient.